control/statesp.py

```diff
@@ -1215,9 +1215,94 @@ def dcgain(self):
 
     def _isstatic(self):
         """True if and only if the system has no dynamics, that is,
-        if A and B are zero. """
+        if `self.A` and `self.B` are zero.
+        """
         return not np.any(self.A) and not np.any(self.B)
 
+    def dynamics(self, t=None, x=None, u=None):
+        """Compute the dynamics of the system
+
+        Given input `u` and state `x`, returns the dynamics of the state-space
+        system. If the system is continuous, returns the time derivative dx/dt
+
+            dx/dt = A x + B u
+
+        where A and B are the state-space matrices of the system. If the
+        system is discrete-time, returns the next value of `x`:
+
+            x[t+dt] = A x[t] + B u[t]
+
+        The inputs `x` and `u` must be of the correct length. The argument
+        `t` is included for consistency with :class:`IOSystem` systems, but
+        is ignored in :class:`StateSpace` systems, which are time-invariant.
+
+        Parameters
+        ----------
+        t : float (ignored)
+            time
+        x : array_like (optional)
+            current state, zero if omitted
+        u : array_like (optional)
+            input, zero if omitted
+
+        Returns
+        -------
+        dx/dt or x[t+dt] : ndarray of shape (self.nstates,)
+        """
+
+        out = np.zeros((self.nstates, 1))
```

Comment:
I might be missing something, but vectors can have ndim=1, and the matrix products and so on will work fine.  Do you need ndim=2 for the vectors here?

Reply:
you're right. I thought there might be ambiguity in how `dot` works on 1D vectors vs column vectors on square arrays, but it looks like it does the right thing for 1D vectors: it interprets them as column vectors. so new PR will return 1D vectors.